terraform/core/46-mwaa-deparments-conn-id-and-permission.tf

```diff
@@ -0,0 +1,125 @@
+/*
+  Create a new IAM user (department_airflow) and store it with the format of mwaa/airflow connection
+  This will be used as aws_conn_id in operators for different departents/teams, e.g. data platform team.
+*/
+
+# Define IAM policies for the permissions of each department's airflow connections
+locals {
+  dataplatform_airflow_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "s3:ListBucket",
+          "s3:GetObject",
+          "s3:PutObject"
+        ],
+        Resource = [
+          module.landing_zone.bucket_arn,
+          "${module.landing_zone.bucket_arn}/*",
+          module.raw_zone.bucket_arn,
+          "${module.raw_zone.bucket_arn}/*",
+          module.refined_zone.bucket_arn,
+          "${module.refined_zone.bucket_arn}/*",
+          module.trusted_zone.bucket_arn,
+          "${module.trusted_zone.bucket_arn}/*",
+          aws_s3_bucket.mwaa_bucket.arn,
+          "${aws_s3_bucket.mwaa_bucket.arn}/*"
+        ]
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "kms:*"
+        ],
+        Resource = "*"
+      },
+      {
+        Effect = "Allow",
+        Action = [
+          "logs:CreateLogStream",
+          "logs:CreateLogGroup",
+          "logs:PutLogEvents",
+          "logs:GetLogEvents",
+          "logs:GetLogRecord",
+          "logs:GetLogGroupFields",
+          "logs:GetQueryResults",
+          "logs:DescribeLogGroups"
+        ],
+        Resource = "arn:aws:logs:*"
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "glue:CreateCrawler",
+          "glue:DeleteCrawler",
```

Comment:
I'd maybe be tempted to remove this action to avoid the opportunity for someone to accidentally remove a crawler with a similar name to the one they actually wanted.

Admittedly, it's an edge case and probably only a theoretical risk anyway.

Reply:
@timburke-hackit  
Since we don't have cases that involve creating crawlers, updating data, and terminating the crawler, as I described a while ago when using a custom EMR operator to do so, we are going to manage the crawler through a Terraform file. Therefore, it would definitely fine to remove both the create and delete permissions.